### PR TITLE
Feat/init build system

### PR DESCRIPTION
**Pull request by xgroleau**
Added nanopb library and an example for future reference. We will probably delete the example once we have actual messages structures. 

For now the CI only checks if the libraries build

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,26 @@
...
-A repo that regroup SwarmUS communication messages with protobuf
+A repo that regroup SwarmUS communication messages with protobuf.
+
+
+## Requirements
+
+* `protoc` to compile .proto files
+* python libraries for protobuf to generate the files
+
+On ubuntu 20.04 you can install it using `apt install protobuf-compiler python3-protobuf`
```
Comment (stlaperriere) [inline README.md:10]: No need for `nanopb`? Or is that already included in some of these dependencies?

**Review by stlaperriere — APPROVED** [review]
Good job! I like the example, it's quite simple to understand.

File: README.md (hunk 1 above)
Comment (g-lesssard) [reply README.md:10]: It is included by CMake in the build

File: example/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+set(LIB_NAME "pheromones-example")
+set(LIB_ALIAS "Pheromones::Example")
```
Comment (g-lesssard) [inline example/CMakeLists.txt:2]: SwarmUs prefix for the lib name and alias?

**Review by g-lesssard — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (xgroleau) [reply README.md:10]: Exaclty, and you can pas NANOPB_PATH if you want to pass the lib from your package manager. (Though we need to build it on embedded anyway)

**Pull request merged**